app/controllers/employee_controller.rb

```diff
@@ -27,6 +27,16 @@ def update
   end
 
   def destroy
+
+    current_week_id = Week.upcoming.first.id
+    assignments = Assignment.where('week_id >= ?', current_week_id).where( employee_id: @employee.id)
```

Comment:
This logic isn't rock solid, which is fine. But, because it might change in the future, I would try to centralize the business logic so that if it does have to change, it only changes in one place.

Please make a scope on the assignment model so that you can do this: `Assignment.upcoming.where(employee: @employee)`. Notice that you'll have to look up the current week in the scope and you don't have to search by employee id (since some Rails magic will work with the defined associations).